tests/run-make/lto-smoke/rmake.rs

```diff
@@ -0,0 +1,15 @@
+// A simple smoke test to check that link time optimization
+// (LTO) works as intended, with its various flags turned on.
+// See https://github.com/rust-lang/rust/issues/10741
```

Comment:
Question: this test is kinda funny. As far as I can tell, this test is only sufficient to establish that:

1. rustc accepts the various `-Clto` flags, and
2. the libraries and binaries are successfully compiled with those flags.

But this test is *not* sufficient to establish that the various LTO flags work -- for all we know, sending LTO flags to rustc could be no-ops and this test can still pass.

(I'm aware this is what the original test does)

Maybe we should instead say that this is a smoke test to check that various LTO flags are accepted and that rustc can successfully compile libraries and binaries with said flags?